File: back-cv/src/cv/entity/cv.schema.ts
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { HydratedDocument } from 'mongoose';
import Activity from 'src/activities/entity/activity.entity';
import Education from 'src/educations/entity/education.entity';
import Experience from 'src/experiences/entity/experience.entity';
import Langue from 'src/langues/entity/langues.entity';
import MadSkill from 'src/mad_skills/entity/mad_skills.entity';
import Profile from 'src/profile/entity/profile.entity';
import Skill from 'src/skills/entity/skills.entity';

@Schema({ timestamps: true })
export class CV {
  @Prop({ type: String, required: true })
  title: string;

  @Prop({ required: false })
  cv_title?: string;

  @Prop({ type: SchemaFactory.createForClass(Profile), required: false, _id: false })
  profile?: Profile;

  @Prop({ required: false })
  tagline?: string;

  @Prop({ type: Number, required: true })
  member_id: number;

  @Prop({ type: [SchemaFactory.createForClass(Experience)], _id: false, required: false })
  experiences: Experience[];

  @Prop({ type: [SchemaFactory.createForClass(Education)], _id: false, required: false })
  educations: Education[];

  @Prop({ type: [SchemaFactory.createForClass(Langue)], _id: false, required: false })
  languages: Langue[];

  @Prop({ type: [SchemaFactory.createForClass(Skill)], _id: false, required: false })
  skills: Skill[];

  @Prop({ type: [SchemaFactory.createForClass(MadSkill)], _id: false, required: false })
  madskills: MadSkill[];

  @Prop({ type: [SchemaFactory.createForClass(Activity)], _id: false, required: false })
  activities: Activity[];
}

export type CvDocument = HydratedDocument<CV>
export const CVSchema = SchemaFactory.createForClass(CV);

File: back-cv-java/src/main/java/com/stever/jobflow/core/CvSchema.java
package com.stever.jobflow.core;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import com.stever.jobflow.core.classes.*;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.List;

@Data
@Document(collection = "cvs")
public class CvSchema {

    @Id
    @JsonProperty("_id")
    private String id;

    @JsonProperty("member_id")
    @Field("member_id")
    private int memberId;

    private String title;

    @Getter(AccessLevel.NONE)
    @Field("cv_title")
    @JsonProperty("cv_title")
    private String cvTitle;

    private String tagline;

    private Profile profile;

    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<Experience> experiences;

    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<Education> educations;

    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<Langue> languages;

    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<Skill> skills;

    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<MadSkill> madskills;

    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<Activity> activities;

    public String getCvTitle() {
        return cvTitle!= null ? cvTitle : title;
    }

}
